cmd/antrea-agent/options.go

```diff
@@ -193,7 +195,7 @@ func (o *Options) setDefaults() {
 	if o.config.OVSRunDir == "" {
 		o.config.OVSRunDir = ovsconfig.DefaultOVSRunDir
 	}
-	if o.config.APIPort == 0 {
+	if !checks.IsValidPort(o.config.APIPort) {
```

Comment:
wouldn't it be better to give an error if the port is not valid (and non-zero), like you do for `tunnelPort` below?